src/globus_sdk/_testing/registry.py

```diff
@@ -0,0 +1,147 @@
+import importlib
+from typing import Any, Dict, Iterator, Optional, Union
+
+import responses
+
+from ..utils import slash_join
+
+
+class RegisteredResponse:
+    _url_map = {
+        "auth": "https://auth.globus.org/",
+        "nexus": "https://nexus.api.globusonline.org/",
+        "transfer": "https://transfer.api.globus.org/v0.10",
+        "search": "https://search.api.globus.org/",
+        "gcs": "https://abc.xyz.data.globus.org/api",
+        "groups": "https://groups.api.globus.org/v2/",
+    }
+
+    def __init__(
+        self,
+        *,
+        path: str,
+        service: Optional[str] = None,
+        method: str = responses.GET,
+        headers: Optional[Dict[str, str]] = None,
+        metadata: Optional[Dict[str, Any]] = None,
+        json: Optional[Dict[str, Any]] = None,
+        body: Optional[str] = None,
+        **kwargs: Any,
+    ) -> None:
+        self.service = service
+        self.path = path
+        if service:
+            self.full_url = slash_join(self._url_map[service], path)
+        else:
+            self.full_url = path
+
+        self.method = method
+        self.json = json
+        self.body = body
+
+        if headers is None:
+            headers = {"Content-Type": "application/json"}
+        self.headers = headers
+
+        self.metadata = metadata or {}
+        self.kwargs = kwargs
+
+    def add(self) -> "RegisteredResponse":
+        kwargs: Dict[str, Any] = {
+            "headers": self.headers,
+            "match_querystring": False,
```

Comment:
There's a bit of a snag here. `0.18.0` fixed this buggy deprecation warning, but it also dropped support for py3.6 .

So for the SDK and CLI, we can't update until we drop 3.6 (which I think we should do soon; I'll queue that up for discussion).

We could try to support `0.17.0` _and_ the later (fixed) versions, and add a test env which runs with the latest `responses`, to make sure it works on both. Otherwise, we need everything which uses this to pin to `responses==0.17.0` until we can drop py3.6 support.
I don't think there are other significant fixes and features in `responses` since `0.17.0`, so pinning appeals a bit for its simplicity.